src/osd/OSDCap.cc

```diff
@@ -116,14 +125,41 @@ bool OSDCapPoolNamespace::is_match_all() const
   return true;
 }
 
+bool OSDCapPoolTag::is_match(const app_map_t& app_map) const
+{
+  if (application.empty()) {
+    return true;
+  }
+  if (app_map.empty()) {
+    return false;
+  }
```

Comment:
I don’t think this logic is a good interface. If we don’t specify an application, we match on anything (because we don’t want to change previously=existing cap behavior, obviously), but if we do specify an application we also need to specify key-value pairs? What if you want to create a “CephFS admin” that can do anything on fs pools? (For something like repair tool nodes. Or rbd admins for running replication agents or something.)

Reply:
Interesting, so would you want something like:

allow rw tag cephfs *=*
or
allow rw tag cephfs *

?

Reply:
Yeah, something like that would be reasonable.